### ISSUE TITLE
[TDThread.FCMacro] Feature Request: Read thread information from hole

### ISSUE THREAD

**Issue by ToniTen** [reox/FreeCAD_macros#1]
Currently, your macro always puts M16 on the label for the dimension. However, if the hole was made with the appropriate tool (part design/hole) it can contain the information on the thread:

![screenshot](https://user-images.githubusercontent.com/55888212/98662521-317fab00-2348-11eb-9438-aefef2437da1.png)

I'm not familiar enough with FreeCAD macros to know how to access the value, but from the GUI, assuming the hole operation is names Hole, you can access it from &lt;&lt;Hole&gt;&gt;.ThreadSize. You also get a boolean &lt;&lt;Hole&gt;&gt;.Threaded to know if it's threaded and so if the value for ThreadSize is meaningful.

**Comment by reox** [comment]
Yes, the macro needs some reworking... But is this not already somehow implemented in TD itself? I just noticed that there are now these extensions: https://wiki.freecadweb.org/TechDraw_Workbench#Extensions
